lib/src/main/java/com/cloudinary/android/preprocess/VideoPreprocessChain.java

```diff
@@ -0,0 +1,44 @@
+package com.cloudinary.android.preprocess;
+
+import android.content.Context;
+import android.net.Uri;
+
+import java.io.File;
+import java.util.UUID;
+
+/**
+ * A preprocess chain to run on videos before uploading. Pass an instance of a populated chain to {@link com.cloudinary.android.UploadRequest#preprocess(PreprocessChain)}.
+ * The processing steps will run by the order in which they were added to the chain. This chain uses the default Video encoder/decoder, however custom implementations
+ * can be supplied using {@link PreprocessChain#loadWith(ResourceDecoder)} and {@link PreprocessChain#saveWith(ResourceEncoder)}.
+ */
+public class VideoPreprocessChain extends PreprocessChain<Uri> {
+
+    private static String targetfilePath;
```

Comment:
No need for a file path - When the chain is created, the original Uri is passed to the first step (via the decoder). Then each step should create a new temp file path to pass on the LiTr, and that's that's the Uri that will get passed on the next step (since it contains the processed file from LiTr). Eventually the encoder will get the last generated file path (via `public String encode(Context context, Uri resource)` and can simply return it as is.